### PR TITLE
.coafile: Add coala bears

### LINKED ISSUE #22: Add .coafile

### PR DESCRIPTION
**Pull request by aashutoshrathi**
Created .coafile file and added bears.

Closes #22

### CODE REVIEW

**Review by RaiVaibhav — CHANGES_REQUESTED** [review]
Always put a new line at EOF

**Review by RaiVaibhav — COMMENTED** [review]
unack aa3c1fdaaeab0040796133d830636a7f05dc981e

remove `for this repository` from commit message and change your commit in that way a person when read the commit message,then understands that a newfile is getting creating here.

**Comment by aashutoshrathi** [comment]
@RaiVaibhav check again

**Review by RaiVaibhav — APPROVED** [review]
As according to issue changes LGTM and also wait for maintainers to review maybe they need more changes

**Comment by jayvdb** [comment]
The .coafile isnt being used in CI, and there are other bears which should be enabled for the contents of this repository.

**Comment by aashutoshrathi** [comment]
This repo mainly has yml and sh files, what else bears should I add?

File: .travis.yml
Comment (RaiVaibhav) [inline .travis.yml]: Does these commits needs to be here

Comment (aashutoshrathi) [reply .travis.yml]: @RaiVaibhav This issue is about adding coala file, but as @jayvdb mentioned, I need to make it work on CI.
Since Travis isn't showing logs of coala. I need help to print it.

Comment (RaiVaibhav) [reply .travis.yml]: @Makman2 @jayvdb  need to know,it need to create `CI Test` regarding `.coafile` like [same as in  coala/coala](https://github.com/coala/coala/blob/9de6241bc8cebcf32bffe773c6ebbb5018bfad6b/tests/coalaCITest.py)

**Comment by RaiVaibhav** [comment]
unack 995fa381ac836e52aa88f28ca9d79a7f2b782b5b
I think you need to create a `CoalaCItest`or something,to check whether `.coafile` exist or not.

File: .travis.yml
Comment (aashutoshrathi) [reply .travis.yml]: The Travis file is all linked with .coafile, it is just not printing output that is given by
```
docker run rultor-python sh -c " "
```
Also, @jayvdb shouldn't we fix Travis link in another PR?

Comment (RaiVaibhav) [reply .travis.yml]: Previously I was saying don't we need to check wherether `.coafile` exists or not???

Comment (RaiVaibhav) [reply .travis.yml]: IMO It  have to check

**Review by RaiVaibhav — CHANGES_REQUESTED** [review]
IMO you have to check `.coafile`

File: .travis.yml
Comment (userzimmermann) [inline .travis.yml]: @aashutoshrathi Hey :) This step has indeed nothing to do with a `.coafile` for this repo itself. It's not even available in the running container. So revert those changes please

**Review by userzimmermann — CHANGES_REQUESTED** [review]
The trick here is that linting the files in this repo has to be done separately from the existing `docker` test steps for container creation from the `Dockerfile`, ideally at the beginning of the `script` section in `.travis.yml`

Just take a look at http://coala.readthedocs.io/en/latest/Users/Docker_Image.html#coala-on-travis-ci to see how this works

cc @jayvdb @RaiVaibhav

**Comment by userzimmermann** [comment]
@aashutoshrathi And BTW: The PR/commit message somehow implies that an existing `.coafile` was extended. I would better name it like `Add .coafile with/for bla bla ...`

But that's up to you ;)

File: .travis.yml
Comment (Monal5031) [reply .travis.yml]: @RaiVaibhav Why do we need to check for the existence of  `.coafile`? 😕

Comment (RaiVaibhav) [reply .travis.yml]: It was my opinion that if we are creating a `.coafile` then we also have to check `.coafile` exist or not 
Its like a enhancement, It will be fine if `Travis CI` also keep a track record of `.coafile` existence of a  `REPO` , its not like a necessity my review was my first comment.

**Comment by userzimmermann** [comment]
@aashutoshrathi Looks better now :) But you still need to run `coala` for actually performing the analysis defined in `.coafile` like shown in http://coala.readthedocs.io/en/latest/Users/Docker_Image.html#coala-on-travis-ci

```yaml
script:
  - docker run -v=$(pwd):/app --workdir=/app coala/base coala --ci
  - ...
```

**Comment by aashutoshrathi** [comment]
@userzimmermann made changes, Review please

**Comment by aashutoshrathi** [comment]
ShellCheckBear is raising these 2 errors : 
```
.ci/check_maintainership.sh
|   7| source·../rultor_github_secrets.sh
|    | [NORMAL] ShellCheckBear:
|    | In POSIX sh, 'source' in place of '.' is undefined. [SC2039]
.ci/check_maintainership.sh
|  10| response=$(curl·"https://api.github.com/teams/1238757/memberships/${author}?access_token=${GITHUB_TOKEN}")
|    | [NORMAL] ShellCheckBear:
|    | author is referenced but not assigned. [SC2154]
```
I need help fixing them.

File: .travis.yml
Comment (jayvdb) [inline .travis.yml]: is it possible it indent the parameter so it doesnt align with the command?

Comment (aashutoshrathi) [reply .travis.yml]: Did you mean?
```
- >
    docker run -v=$(pwd):/app -v=/tmp/coala:/cache
      --workdir=/app coala/base coala --ci
```
Not possible as the command doesn't work this way.

**Comment by aashutoshrathi** [comment]
@jayvdb @userzimmermann review again

File: .travis.yml
Comment (jayvdb) [inline .travis.yml]: why is `-v=/tmp/coala:/cache` here?

Comment (userzimmermann) [reply .travis.yml]: @aashutoshrathi @jayvdb I'm wondering too :) It's not necessary

File: README.md
Comment (userzimmermann) [inline README.md:2]: @aashutoshrathi Please put this in a separate commit before the main commit. It's necessary to make this PR pass. But it's actually not part of adding a `.coala` file and a `coala` test step. Just to keep things clean

File: .travis.yml
Comment (userzimmermann) [reply .travis.yml]: @aashutoshrathi Without the unnecessary `-v=/tmp/coala:/cache`, this command can be written as a one-liner

**Review by userzimmermann — CHANGES_REQUESTED** [review]
@aashutoshrathi Good work :) Just a little more cleanup and this PR can be merged

File: .travis.yml
Comment (aashutoshrathi) [reply .travis.yml]: No need here, fixing it.

File: README.md
Comment (aashutoshrathi) [reply README.md:2]: I changed the PR message to 
```
Created .coafile file and added bears
Fixed CI integration with coala.
Fixed errors raised by bears.
```

**Comment by aashutoshrathi** [comment]
Review again

**Review by userzimmermann — APPROVED** [review]

**Comment by userzimmermann** [comment]
ack https://github.com/coala/rultor-python/pull/27/commits/f4cf228ec2b18410e01b190cdf7e069d82238a22

**Comment by userzimmermann** [comment]
@rultor merge

**Comment by userzimmermann** [comment]
@aashutoshrathi Done! Thanks :)

File: .travis.yml
Comment (jayvdb) [inline .travis.yml:6]: This would have fit on a single line.

Comment (Makman2) [reply .travis.yml:6]: Filed https://github.com/coala/rultor-python/issues/32 :+1:

Comment (userzimmermann) [reply .travis.yml:6]: @jayvdb @Makman2 Just didn't want to insist too much ;)

@aashutoshrathi But I told you in https://github.com/coala/rultor-python/pull/27#discussion_r148964949 so go ahead to your next PR for https://github.com/coala/rultor-python/issues/32 :)

**Pull request merged**
